packages/project/src/Workflow.ts

```diff
@@ -167,6 +169,18 @@ class Workflow {
   getVersionHash() {
     return generateHash(this);
   }
+
+  pushHistory(versionHash: string) {
+    this.history.push(versionHash);
+  }
+
+  // return true if the current workflow can be merged into the target workflow without losing any changes
+  canMergeInto(target: Workflow) {
+    if (!target.history.length) return true;
+    const targetHead = target.history[target.history.length - 1];
```

Comment:
Open question: is the head the last history, or whatever the current actual version is?